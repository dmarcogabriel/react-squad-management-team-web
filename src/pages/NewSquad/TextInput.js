import React from 'react';
import cn from 'classnames';

const TextInput = ({
  label,
  value,
  id,
  error,
  onChange = () => {},
  ...props
}) => (
  <div>
    <label htmlFor={id}>
      <p className={cn('font-bold text-base', error && 'text-primary-dark ')}>
        {label}
      </p>
      <input
        id={id}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        className={cn(
          'border-2 rounded p-1 focus:border-gray-500',
          error && 'border-primary-dark focus:border-primary-dark'
        )}
        {...props}
      />
    </label>
  </div>
);

export default TextInput;
