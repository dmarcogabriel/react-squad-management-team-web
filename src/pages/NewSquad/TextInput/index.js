import React from 'react';
import cn from 'classnames';

const TextInput = ({
  label,
  value,
  id,
  error,
  onChange,
  className = '',
  dataTestid,
  labelTestid,
  ...props
}) => (
  <div className="w-full">
    <label htmlFor={id}>
      <p
        data-testid={labelTestid || 'label'}
        className={cn('font-bold text-base', error && 'text-primary-dark')}
      >
        {label}
      </p>
      <input
        data-testid={dataTestid || 'input'}
        id={id}
        value={value}
        onChange={(e) => onChange(e.target.value)}
        className={cn(
          'border-2 w-full rounded p-1 focus:border-gray-500',
          error && 'border-primary-dark focus:border-primary-dark',
          className
        )}
        {...props}
      />
    </label>
  </div>
);

export default TextInput;
