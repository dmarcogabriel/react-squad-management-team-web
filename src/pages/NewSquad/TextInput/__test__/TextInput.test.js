import { render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import userEv from '@testing-library/user-event';
import TextInput from '..';

describe('<TextInput />', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TextInput />, div);
  });

  it('render label correctly', () => {
    const { getByTestId } = render(<TextInput label="Test" />);

    const label = getByTestId('label');

    expect(label).toHaveTextContent('Test');
  });

  it('render lable error highlight correctly', () => {
    const { getByTestId } = render(<TextInput label="Test" error />);

    const label = getByTestId('label');

    expect(label.className).toEqual('font-bold text-base text-primary-dark');
  });

  it('render input border error highlight correctly', () => {
    const { getByTestId } = render(<TextInput error />);

    const input = getByTestId('input');

    expect(input.className).toEqual(
      'border-2 w-full rounded p-1 focus:border-gray-500 border-primary-dark focus:border-primary-dark'
    );
  });

  it('execute onChange event', () => {
    let result;
    const onChange = (e) => {
      result = e;
    };

    const { getByTestId } = render(<TextInput onChange={onChange} />);

    userEv.type(getByTestId('input'), 'Testing');

    expect(result).toEqual('Testing');
  });

  it('match snapshot', () => {
    expect(shallow(<TextInput />)).toMatchSnapshot();
  });
});
