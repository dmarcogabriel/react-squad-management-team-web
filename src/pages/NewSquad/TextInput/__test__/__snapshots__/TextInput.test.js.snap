// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> match snapshot 1`] = `
<div
  className="w-full"
>
  <label>
    <p
      className="font-bold text-base"
      data-testid="label"
    />
    <input
      className="border-2 w-full rounded p-1 focus:border-gray-500"
      data-testid="input"
      onChange={[Function]}
    />
  </label>
</div>
`;
